src/utils/file-explorer.cpp

```diff
@@ -24,8 +24,12 @@ namespace pinocchio
       while(true)
       {
         size_t offset = policyStr.find_first_of(delimiter, lastOffset);
-        if (offset < policyStr.size())
+        if (offset < policyStr.size()) {
           list_of_paths.push_back(policyStr.substr(lastOffset, offset - lastOffset));
+          // Support for devel spaces: We also need to look one package above:
+          // Work-around for https://github.com/stack-of-tasks/pinocchio/issues/1463
+          list_of_paths.push_back(policyStr.substr(lastOffset, offset - lastOffset) + "/..");
+        }
```

Comment:
This is a bit hazardous and kind of intrusive...

I would choose instead modify `rosPaths` function (and make it available in Python) as you did here or, arguably better, as:
```cpp
std::vector<std::string> new_list;
for (auto path : list_of_paths) {
  new_list.push_back(path);
  if (file_exists(path + "/package.xml"))
    new_list.push_back(path + "/..");
}
```

Reply:
Not sure whether I wasn't clear or whether you think this is a bad idea so I reformulate my proposition.

I think it is better to fix the list of ROS package paths in function `rosPaths` and not in this function so that it does not apply to anything else than ROS.
Someone later may use it's own environment variable and definitely won't expect this way of resolving paths.

Reply:
Thanks for the comment, I am modifying it now